### ISSUE TITLE
DeepAREstimator unexpected 'target_dim'

### ISSUE THREAD

**Issue by Narsiss** [awslabs/gluon-ts#157]
## Description
I'm trying DeepAREstimator to predict 4 dimensions data.
This address shows a parameter name 'target_dim' to set  dimension of the target.
But when I create the estimator like this:
estimator = DeepAREstimator(freq="1D",
                            prediction_length=20,
                            target_dim=4,
                            trainer=Trainer(epochs=15))
Got an error: __init__() got an unexpected keyword argument 'target_dim'

Please help

**Comment by Narsiss** [comment]
This is a screenshot of my data:
![屏幕快照 2019-07-01 下午5 22 50](https://user-images.githubusercontent.com/5062214/60425386-f97d3980-9c24-11e9-8a6c-14f35806bbc2.png)

**Comment by mbohlkeschneider** [comment]
Hi. The `target_dim` parameter is a leftover from some work-in-progress that we forgot in the code. 

You can just pass the dataset into DeepAR without setting a dimensionality parameter. Please refer to the [documentation](https://github.com/awslabs/gluon-ts) for information on how to prepare the dataset. 

Note that although DeepAR expects a set of univariate time series, it will still use patterns across the time series to make forecasts.

**Comment by lostella** [comment]
Hi @Narsiss, the `target_dim` constructor argument is not required, and we will remove it from the documentation.

To deal with 4-dimensional data you need however to specify an appropriate data distribution that the model should use: you can try setting `distr_output=MultivariateGaussianOutput(dim=4)` to start with. Or, you can provide your 4 time series as independent entries in the dataset, and let the `DeepAREstimator` use the default (univariate) Student-t distribution as output of the model.

**Comment by Narsiss** [comment]
> prepare

Thanks

**Comment by Narsiss** [comment]
> Hi @Narsiss, the `target_dim` constructor argument is not required, and we will remove it from the documentation.
> 
> To deal with 4-dimensional data you need however to specify an appropriate data distribution that the model should use: you can try setting `distr_output=MultivariateGaussianOutput(dim=4)` to start with. Or, you can provide your 4 time series as independent entries in the dataset, and let the `DeepAREstimator` use the default (univariate) Student-t distribution as output of the model.
-----------------------------------------

I've think about trained 4 time series, but the 4-dimensional data maybe influence each other.
I will try to use distr_output=MultivariateGaussianOutput(dim=4) . 

Thanks for you help

**Comment by Narsiss** [comment]
@lostella 
I'm trying to use MultivariateGaussianOutput like this:
![屏幕快照 2019-07-01 下午6 11 04](https://user-images.githubusercontent.com/5062214/60428569-b4103a80-9c2b-11e9-8153-dd82db14a984.png)
But the train cannot start. 
Please help, thanks

**Comment by lostella** [comment]
@Narsiss one problem you have is that the axis of `target` are inverted: in the dataset, you should provide the covariate dimension as first axis, and time as second axis. To be clear, a dataset like the following should work:

```
train_dataset = ListDataset(
    data_iter=[
        {
            "start": "2019-01-01 00:00:00",
            "target": np.ones(shape=(4, 100)).tolist(),
        },
    ],
    freq="5min",
)
```

However I'm encountering problems with this, so I'll open a separate issue about `MultivariateGaussian`. Can you try using univariate distribution (like the default one) and model your data as four independent time series? To make sure that there's no other issue with your data.

**Comment by lostella** [comment]
@Narsiss I've opened #160 to track the issues with multivariate data.

As a suggestion: when opening issues, could you please include minimal code examples that raise the error you get, and that one can easily copy paste to reproduce? Thanks

**Comment by Narsiss** [comment]
@lostella thank you

**Comment by Joachim96** [comment]
@Narsiss when you say that the training cannot start, do you receive an error? Or is it simply stuck at 0%?

I've been playing a bit with DeepAREstimator recently and I sometimes have to wait for 30-60 minutes (with the same display as what you showed) before the training actually shows some progress.

I don't pretend to be able to understand or correct anything that might go wrong in here but I just wanted to tell you that if you have no error message, patience might just do the trick.

**Comment by Narsiss** [comment]
@Joachim96  stuck at 0%

**Comment by Narsiss** [comment]
@Joachim96  stuck at 0%
Just like the pictures before. I will change my code as lostella's code when the #160 fix.

**Comment by lostella** [comment]
Closing since this is tracked in #160
